### PR TITLE
Use whoami instead of $USER

### PR DESCRIPTION
**Pull request by gerashegalov**
Docker does not create the envrironment variable USER per https://docs.docker.com/engine/reference/run/#environment-variables

The user property in the build-info is missing as seen in #1840 

Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

**Comment by gerashegalov** [comment]
build

File: build/build-info
Comment (jlowe) [inline build/build-info:1]: 2024 copyrights

Comment (gerashegalov) [reply build/build-info:1]: will wait until #1840 is merged to save CI resources

**Review by jlowe — APPROVED** [review]

**Pull request merged**
